### PR TITLE
Enable Multiple Anvils for Isherwood Mission

### LINKED ISSUE #77908: Isherwood mission for an anvil does not accept craftable anvils (i.e small anvil) despite suggesting crafting one.
### Describe the bug

Isherwood mission for an anvil does not accept craftable anvils (i.e small anvil) despite suggesting crafting one when you ask for advice. (also real anvils are rare enough to make the mission practically uncompletable if we actually demand one...)

### Attach save file

N/A

### Steps to reproduce

try to turn in a small anvil, be refused.

### Expected behavior

if the mission advices you to craft an anvil it should accept craftable anvils. 

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22631.4460 (23H2)
- Game Version: cdda-experimental-2024-11-15-0512 0f2c155 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Bombastic Perks [bombastic_perks],
    Bionic Professions [package_bionic_professions]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TheMurderUnicorn**
#### Summary
Bugfixes "Enable Other Anvils for Isherwood Quest"

#### Purpose of change
#77908 

TLDR: Missions asks for an anvil, dialogue suggests that you can use a crafted one.  However, it only accepted the regular anvil.  It was requested that this change to accommodate other anvils.

#### Describe the solution
Restructure mission using if then logic to allow for the anvil, crude anvil, bronze anvil, or heavy anvil.

#### Describe alternatives you've considered
Removing the dialogue suggesting craft-able anvils are acceptable and leaving it as anvil only.

#### Testing
Ensured no syntax errors.  Verified that all anvils are acceptable/complete the mission (though it requires you to wield that larger anvils/have them in your inventory to complete, this is a flaw with the removal command, I can't fix that).  Verified the anvil is removed before giving the reward, I don't know if this matters, but it makes more sense that way and figured it was safer that way.

#### Additional context

Okay, so a few issues.  Firstly, all anvils have anvil 3 or better, making them suitable for the purpose of this quest.  Therefore, I feel they should be acceptable.  However, part of this mission updates the mapgen to include that anvil as furniture IN the map.  I looked into changing this to accommodate which anvil you received, however I could not find an example of "then" using the update mapgen command.  When attempting to do this I got various errors about it not being a valid entry for this.  I don't know if I was doing it wrong or if the infrastructure isn't there.

TLDR: TODO: Figure out how to change the mapgen update to reflect which anvil is turned in for the quest.

Additionally, it was unclear to me if the bronze anvil should be valid, but seeing as we assign them the same anvil quality, presumably it is fine.
closes #77908 

Screenshots:
image
image
image

### CODE REVIEW

**Comment by EvanBalster** [comment]
Related: https://github.com/CleverRaven/Cataclysm-DDA/issues/77866
